filters/cookie/cookie_test.go

```diff
@@ -382,3 +383,342 @@ func TestSetCookie(t *testing.T) {
+			if err != nil {
+				t.Fatalf("Failed to create filter: %v", err)
+			}
+
+			if f == nil {
+				t.Fatal("Failed to create filter: filter nil")
+			}
+
+			ctx := &filtertest.Context{
+				FRequest: &http.Request{
+					Header: tt.cookies,
+					Host:   "foo"},
+				FStateBag: map[string]interface{}{},
+				FResponse: &http.Response{Header: http.Header{}},
+			}
+
+			f.Request(ctx)
+
+			if c, err := ctx.Request().Cookie(tt.arg); err != http.ErrNoCookie {
+				t.Fatalf("Failed to delete cookie %s: %q", tt.arg, c)
+			}
+
+			for k, a := range tt.wantCookies {
+				cookie, err := ctx.Request().Cookie(k)
+				if err != nil {
+					t.Fatalf("Failed to get cookie %q: %v", k, err)
+				}
+
+				v := a[0]
+				if cookie.Value != v {
+					t.Fatalf("Failed to get cookie value %q, got: %q", v, cookie.Value)
+				}
+			}
+
+			for _, cookie := range ctx.Request().Cookies() {
+				if a, ok := tt.wantCookies[cookie.Name]; !ok {
+					t.Fatalf("Failed to delete cookie: %s", cookie.Name)
+				} else {
+					found := false
+					for _, v := range a {
+						if v == cookie.Value {
+							found = true
+							break
+						}
+					}
+					if !found {
+						t.Fatalf("Failed to get the expected cookie value, got: %s", cookie.Value)
+					}
+
+				}
+			}
+		})
+	}
+
+}
+
+func TestDropResponseCookie(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		arg         string
+		cookies     http.Header
+		wantCookies map[string][]string
+	}{
+		{
+			name:        "test no cookies",
+			arg:         "no-cookie",
+			cookies:     nil,
+			wantCookies: nil,
+		},
+		{
+			name: "test one cookie not match",
+			arg:  "no-match",
+			cookies: http.Header{
+				"Set-Cookie": []string{
+					"foo=foo1",
+				},
+			},
+			wantCookies: map[string][]string{"foo": {"foo1"}},
+		},
+		{
+			name: "test one cookie with match",
+			arg:  "foo",
+			cookies: http.Header{
+				"Set-Cookie": []string{
+					"foo=foo1",
+				},
+			},
+			wantCookies: nil,
+		},
+		{
+			name: "test two cookies and one cookie with match",
+			arg:  "foo",
+			cookies: http.Header{
+				"Set-Cookie": []string{
+					"foo=foo1",
+					"bar=baz",
+				},
+			},
+			wantCookies: map[string][]string{"bar": {"baz"}},
+		},
+		{
+			name: "test two cookies with the same name and no match",
+			arg:  "foo",
+			cookies: http.Header{
+				"Set-Cookie": []string{
+					"bar=foo1",
+					"bar=baz",
+				},
+			},
+			wantCookies: map[string][]string{"bar": {"foo1", "baz"}},
+		},
+		{
+			name: "test two cookies with the same name and both match match",
+			arg:  "foo",
+			cookies: http.Header{
+				"Set-Cookie": []string{
+					"foo=foo1",
+					"foo=baz",
+				},
+			},
+			wantCookies: nil,
+		},
+		// {
+		// 	name: "test one multivalue cookie with 2 names and no match",
+		// 	arg:  "foo",
+		// 	cookies: http.Header{
+		// 		"Set-Cookie": []string{
+		// 			"bar=foo1;qux=baz",
+		// 		},
+		// 	},
+		// 	wantCookies: map[string][]string{"bar": {"foo1"}, "qux": {"baz"}},
+		// },
```

Comment:
@AlexanderYastrebov any idea why this would not work?
I get only cookie "bar" no "qux"...

Reply:
Set-Cookie header format differs from Cookie header format - it does not allow multiple cookies per line but uses semicolon as attribute separator.
See
* https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
* https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie

The dropResponseCookie test should be changed to use multiline values like
```go
"Set-Cookie": []string{
	"bar=foo1",
	"qux=baz",
},
```

Reply:
Oh, good that you checked it